examples/data-objects/primitives/src/main.tsx

```diff
@@ -43,7 +43,7 @@ export class PrimitivesCollection extends DataObject implements IFluidHTMLView {
     public render(div: HTMLElement) {
         const mapCreate = (name: string) => SharedMap.create(this.runtime, name);
         const mapListen = (listener: (changed: IDirectoryValueChanged) => void) => {
-            this.root.on("valueChanged", (changed) => {
+            this.mapDir.on("containedValueChanged", (changed) => {
                 if (changed.path !== this.mapDir.absolutePath) {
                     return;
                 }
```

Comment:
And since the event is specific to the `mapDir`, we don't need to check the path anymore -- we know it relates to the correct subdirectory already.

```suggestion
```

Reply:
Made the proposed changes @ChumpChief